File: 2271-rearrange-array-elements-by-sign/2271-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        int n = nums.size();
        vector<int> result(n);
        int posIndex = 0, negIndex = 1;

        // Traverse the array and place positive and negative numbers
        for (int i = 0; i < n; i++) {
            if (nums[i] > 0) {
                result[posIndex] = nums[i];
                posIndex += 2;  // Place positive numbers at even indices
            } else {
                result[negIndex] = nums[i];
                negIndex += 2;  // Place negative numbers at odd indices
            }
        }
        return result;
    }
};

File: 2271-rearrange-array-elements-by-sign/2271-rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums):
        n = len(nums)
        result = [0] * n
        posIndex, negIndex = 0, 1

        # Traverse the array and place positive and negative numbers
        for num in nums:
            if num > 0:
                result[posIndex] = num
                posIndex += 2  # Place positive numbers at even indices
            else:
                result[negIndex] = num
                negIndex += 2  # Place negative numbers at odd indices

        return result